File: app/src/main/java/com/ivanovsky/passnotes/data/repository/db/converters/FSTypeConverter.java
package com.ivanovsky.passnotes.data.repository.db.converters;

import androidx.room.TypeConverter;

import com.ivanovsky.passnotes.data.repository.file.FSType;

public class FSTypeConverter {

    @TypeConverter
    public FSType fromDatabaseValue(String value) {
        return FSType.findByValue(value);
    }

    @TypeConverter
    public String toDatabaseValue(FSType fsType) {
        return fsType != null ? fsType.getValue() : null;
    }
}

File: app/src/main/java/com/ivanovsky/passnotes/data/repository/db/converters/FSTypeConverter.kt
package com.ivanovsky.passnotes.data.repository.db.converters

import androidx.room.TypeConverter
import com.ivanovsky.passnotes.data.entity.FSType
import com.ivanovsky.passnotes.data.entity.FSType.Companion.findByValue

object FSTypeConverter {

    @TypeConverter
    fun fromDatabaseValue(value: String): FSType {
        return findByValue(value) ?: FSType.REGULAR_FS
    }

    @TypeConverter
    fun toDatabaseValue(fsType: FSType): String {
        return fsType.value
    }
}